src/libmongoc/src/mongoc/mongoc-crypt.c

```diff
@@ -628,14 +637,41 @@ _needs_on_demand_aws_kms (bson_t const *kms_providers, bson_error_t *error)
    }
 }
 
+/**
+ * @brief Check whether the given kmsProviders object requests automatic Azure
+ * credentials
+ *
+ * @param kmsprov The input kmsProviders that may have an "azure" property
+ * @param error An output error
+ * @retval true If success AND `kmsprov` requests automatic Azure credentials
+ * @retval false Otherwise. Check error->code for failure.
+ */
+static bool
+_check_azure_kms_auto (const bson_t *kmsprov, bson_error_t *error)
+{
+   bson_set_error (error, 0, 0, " ");
```

Comment:
```suggestion
   *error = (bson_error_t){0};
```

Reply:
bson_set_error handles null pointers safely.

Reply:
Point. Should `" "` be `""`?
```suggestion
   bson_set_error (error, 0, 0, "");
```

Reply:
You'd think so, but GCC warns that empty format strings are a non-standard C extension (??)

Reply:
I suppose `bson_set_error (error, 0, 0, "%s", "");` would be a way to satisfy GCC, although I am not very happy about it.

Avoiding the string literal was what motivated my initial suggestion; a simple null check may suffice:
```c
   if (error) {
      *error = (bson_error_t){0};
   }
```
I'd prefer not use `bson_set_error` to "reset" an error object. I'll leave it to you to decide which is preferable.